src/olympia/blocklist/mlbf.py

```diff
@@ -127,34 +142,38 @@ def not_blocked_json(self):
         with storage.open(self._not_blocked_path, 'r') as json_file:
             return json.load(json_file)
 
+    def fetch_not_blocked_json(self):
+        """A safer version of .not_blocked_json that will generate from the db
+        if the cached_property isn't set."""
+        try:
+            return self.not_blocked_json
+        except FileNotFoundError:
+            # when self._not_blocked_path doesn't exist
+            pass
+        # see fetch_blocked_json
```

Comment:
This feels a bit ugly. The fact that you need this comment signals to me that these functions are coupled in a way that is non-intuitive. I wonder if a shared helper would make thus a bit cleaner?

Reply:
before I had a cached_property around `fetch_blocked_json` but I stripped it out because cached properties relying on cached properties tends to end up with annoying end cases. (though as we don't change them after being set here that could work still...)